### PR TITLE
[#157308219] fix: superagent mock

### PR DESCRIPTION
**Pull request by gunzip**
Actual implmentation is clumsy and not safe to run between tests (mutates global state)

Replaced with this two line implementation: 
https://github.com/teamdigitale/digital-citizenship-functions/pull/209/files#diff-745f8e09108272a03a8fb116f2f16e7bR16

which is also more flexible.

### CODE REVIEW

**Comment by digitalcitizenship** [comment]
## Affected stories

  * 🐞 [#157308219](https://www.pivotaltracker.com/story/show/157308219): Fix superagent mock

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: lib/utils/__tests__/mailup.test.ts [hunk 1]
```diff
@@ -4,37 +4,20 @@ import Mail = require("nodemailer/lib/mailer");
...
-  (superagent as any).__resetMocks();
-};
+afterEach(() => {
+  jest.restoreAllMocks();
+  jest.resetAllMocks();
+});
+
+const mockSuperagentResponse = (response: any) =>
+  // tslint:disable-next-line:no-object-mutation
+  (Object.getPrototypeOf(request("", "")).send = jest.fn()).mockReturnValueOnce(
```
Comment (cloudify) [inline lib/utils/__tests__/mailup.test.ts]: I can't even understand what this does, please explain it very well :)

Comment (gunzip) [reply lib/utils/__tests__/mailup.test.ts]: added docs

File: lib/utils/__tests__/mailup.test.ts [hunk 2]
```diff
@@ -13,9 +13,13 @@ afterEach(() => {
...
 });
 
+// as superagent does not export request methods directly
+// we must override the superagent.Request prototype
+// so we can set up our jest mock to use it instead
+// of the send() method
 const mockSuperagentResponse = (response: any) =>
   // tslint:disable-next-line:no-object-mutation
-  (Object.getPrototypeOf(request("", "")).send = jest.fn()).mockReturnValueOnce(
+  ((superagent as any).Request.prototype.send = jest.fn()).mockReturnValueOnce(
```
Comment (cloudify) [inline lib/utils/__tests__/mailup.test.ts]: thx, I would just split this line for readability:

```ts
const sendMock = jest.fn();
(superagent as any).Request.prototype.send = sendMock;
sendMock.mockReturnValueOnce(...)
```

Comment (gunzip) [reply lib/utils/__tests__/mailup.test.ts]: another option is to just use
```ts
const getSendMock = () => (superagent as any).Request.prototype.send
```

then use it inside tests with 
```
getSendMock().mockReturnValue(...)
```

Comment (cloudify) [reply lib/utils/__tests__/mailup.test.ts]: `getSendMock` can be misleading as it creates a new mock each time, perhaps `newSendMock` or `initSendMock`

anyway, just make the code cleaner by splitting the creation of the mock from its usage

Comment (gunzip) [reply lib/utils/__tests__/mailup.test.ts]: done

**Pull request merged**